### ISSUE TITLE
Comparing of two faces

### ISSUE THREAD

**Issue by apophatique** [nizhib/pytorch-insightface#7]
Hi! Thanks for your work, it's very simple to use

I try to compare two faces on images. Could you please tell me how to property compare one face to another after getting features? 

I pass to the script two faces:
![image](https://user-images.githubusercontent.com/44003762/116057831-0217ed00-a6a1-11eb-8ea8-53d0e07a315a.png)
![image](https://user-images.githubusercontent.com/44003762/116057855-08a66480-a6a1-11eb-9e0d-eafa99f5c232.png)

Then i do resize it to 112/112, and after np.dot(x1, x2) i'm getting 0.93235177. What is the threshold for comparing two people in the photo?

Here is my code:
 
    embedder = insightface.iresnet100()
    embedder.eval()
  
    mean = [0.5] * 3
    std = [0.5 * 256 / 255] * 3
    preprocess = transforms.Compose([
          transforms.Resize((112, 112)),
          transforms.ToTensor(),
          transforms.Normalize(mean, std)
    ])
  
    face1 = Image.open('FaceX-Zoo/face_sdk/database/BradPitt.png')
    face2 = Image.open('FaceX-Zoo/face_sdk/database/Margot Robbie.png')
  
    if __name__ == '__main__':
      
          tensor1 = preprocess(face1)
          tensor2 = preprocess(face2)
      
          with torch.no_grad():
              start = time.time()
              features1 = embedder(tensor1.unsqueeze(0))[0]
              features2 = embedder(tensor2.unsqueeze(0))[0]
      
              feature1 = features1.cpu().numpy()
              feature2 = features2.cpu().numpy()
              x1 = feature1 / np.linalg.norm(feature1)
              x2 = feature2 / np.linalg.norm(feature2)
      
              p = np.dot(x1, x2)
              print(p)

Maybe the problem is in 112x112 resolution? If it is, how it is possible to pass larger images?

Thanks!

**Comment by nizhib** [comment]
Hi @apophatique! The repo contains only the embedding part and the model expects the input to be prepared in a standard "detect a face with 5 keypoints, align and crop the face" way like in https://github.com/urbaneman/Face_crop_align_mtcnn (haven't checked the code there, but at least you can understand what is expected to be done before feeding the model).

The sample image in the repo is already prepared that way what is unfortunately misleading.

If you do it correctly, the common thresholds are somewhere between 0.5-0.7 i.e. dist < 0.5 => the same person.
